### PR TITLE
Migration/l10n br crm

### PR DESCRIPTION
**Pull request by renatonlima**

### CODE REVIEW

**Comment by mileo** [comment]
@renatonlima are you redoing the work of https://github.com/OCA/l10n-brazil/pull/124 ?

**Comment by rvalyi** [comment]
so once this branch with l10n_br_base https://github.com/OCA/l10n-brazil/pull/129 is passing tests again, it needs to be merged into this one to enable the installation and testing by Travis.

**Comment by rvalyi** [comment]
So I didn't merge l10n_br base in this branch to avoid possible conflicts when merging globally. Instead here is the test branch where we also have code-metrics: https://github.com/OCA/l10n-brazil/pull/154

The coverage is lower with the l10n_br_crm module and I think it deserved a dumb lead conversion test before we merge.
Lint results are here: https://travis-ci.org/OCA/l10n-brazil/jobs/66147684
We need them green before considering merge, so :-1: for now.

@mileo. Same as for the l10n_br_base PR. Please if we are missing any good feature you guys did, please cherry pick the commits and rebase them on this branch We will do our best to look at your work before merging anyway. Sorry again for the inconvenience of the double work here, again explained by the very exceptional conditions we had.

File: l10n_br_crm/models/__init__.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# -*- encoding: utf-8 -*-
+# -*- coding: utf-8 -*-
 ###############################################################################
 #                                                                             #
-# Copyright (C) 2011  Renato Lima - Akretion                                  #
+# Copyright (C) 2015  Renato Lima - Akretion                                  #
```
Comment (rvalyi) [inline l10n_br_crm/models/__init__.py]: should be 2011 - TODAY

**Comment by rvalyi** [comment]
:+1: 
PEP is OK, installation works on the test branch.
We can eventually add a lead conversion test.

**Comment by rvalyi** [comment]
@mileo I see that in your branch you added the rg on a lead (seems legitimate), you also separated the cpf and cnpj fields. Could you elaborate on why (it's not in the commit message I think)? @renatonlima what do you think about this design? https://github.com/odoo-brazil/l10n-brazil/blob/8.0/l10n_br_crm/crm_lead.py#L42 cc @danimaribeiro

**Comment by rvalyi** [comment]
OK, so we are now green on tests and PEP-8.
That's sad to decrease the coverage, so that would be cool to add some basic lead conversion test and keep a high coverage.

@mileo we found some design changes you did here like separating the cpf and cnpj fields. Could you please argue why you think it's better than what we have now? @renatonlima didn't seem convinced but may be you'll convince us.

**Comment by danimaribeiro** [comment]
One is used for the company data, and the other for the contact. 
In this case, I think is necessary to have both fields.

capturar

Not ok to merge yet.

**Comment by danimaribeiro** [comment]
It happens an error when converting a lead to an oportunity.

error creating lead

The method signature is different from the crm base class. Adjusting the signature to the following:

new signature

It works partially, because now I get the error about "CNPJ invalido":
cnpj invalido

So definitely we need two different fields, CNPJ and CPF, one is for legal_name and the other for contact_name.

**Comment by danimaribeiro** [comment]
It's better to get the code from odoo-brazil, with @api.model, the lead conversion to opportunity is working nicely, generating both company and contact with the corresponding CNPJ and CPF.

It would be nice to have the address fields order according to partner and company, but that I can submit a patch after.

**Comment by danimaribeiro** [comment]
Merged in #165

**Comment by rvalyi** [comment]
oups that one had to be closed then

**Pull request closed without merging**